### ISSUE TITLE
implement monitoring for MDN

### ISSUE THREAD

**Issue by metadave** [mozmeao/infra#398]
- New Relic synthetics for overall site performance
- deadmanssnitch for any backup cron tasks
- CloudWatch?

**Comment by metadave** [comment]
https://github.com/mozmeao/infra/issues/526

**Comment by jwhitlock** [comment]
@metadave that covers the New Relic piece. Do you want to go over other bits on Wednesday? Maybe we can prototype on staging.

**Comment by metadave** [comment]
- new DeadmansSnitch setup: "MDN Prod S3 backup cron"
- incoming PR for DeadmansSnitch/mdn prod backup

**Comment by metadave** [comment]
https://github.com/mozmeao/infra/pull/533

**Comment by metadave** [comment]
https://github.com/mozmeao/infra/pull/542
https://github.com/mozmeao/infra/pull/547
https://github.com/mozmeao/infra/pull/548

**Comment by jwhitlock** [comment]
And now #549, which I thinks closes out the New Relic bit.

**Comment by jwhitlock** [comment]
I think the only missing monitoring is the size of the celery queues, and I'm not sure that should be a blocker.

SCL3 uses RabbitMQ. On the RabbitMQ server, you can request the depth of the queues.  If they exceed a big number (10k? 50k?), an alert is raised in Nagios, and the MOC is informed.  This has caused problems in the past, since the ``mdn_purgable`` queue takes tasks that scale with the number of pages on MDN, and can get filled up by a spider.

When redis is the backend, Celery creates a list named after the queue, such as ``mdn_purgable``.  To get the depth of the queue, you'd use the command "[llen](https://redis.io/commands/llen) mdn_purgable", and count a missing key as 0.

[Datadog has a Redis integration](https://docs.datadoghq.com/integrations/redisdb/), with a mechanism for reporting on the length of select keys:

* [warn_on_missing_keys](https://github.com/Datadog/integrations-core/blob/master/redisdb/conf.yaml.example#L45) - Defaults to ``True``, we'd want ``False``, since the queues are not created until used.
* [keys](https://github.com/Datadog/integrations-core/blob/master/redisdb/conf.yaml.example#L38-L40) - We'd want to set this to the list of queues, ``mdn_purgeable``, ``mdn_search``, ``mdn_emails``, ``mdn_wiki``, and ``celery``.

But, we're not running Redis directly, we're using ElastiCache.  [Datadog integrates with ElastiCache](https://docs.datadoghq.com/integrations/awselasticache/), and requires some additional config to tell it what engine you're using, and how to connect with it.  However, it looks like it uses the Redis configuration method, so we may have access to the same ``llen`` functionality.

@metadave, do we have a system where we could run the Datadog ElastiCache integration? This would give us superior monitoring of the celery queues, with tracking of queue depths over time.

**Comment by jwhitlock** [comment]
I thought of the one more thing - spliting alerts so devs get backend and non-critical alerts, and devs + SREs get user-facing critical alerts. I think that's creating and tuning the New Relic alert policies.

**Comment by metadave** [comment]
@jwhitlock we use DaemonSets to run the Datadog agents, meaning that every node in the cluster runs an agent (including the master). Instead of modifying the DaemonSet config, we can create a tiny container that runs the DD agent + [Elasticache integration](https://docs.datadoghq.com/integrations/awselasticache/).

**Comment by jwhitlock** [comment]
After an IRC discussion, I created a New Relic alert policy "MDN Dev Alerts", that alerts me and the MDN admins, and assigned the "backend" apps to it (``kuma-backend-prod-portland``, ``kumascript-stage-portland``, etc.).  I assigned the "frontend" apps (``kuma-web-prod-portland`` and ``kuma-web-stage-portland``) to the existing "MDN Alerts" policy, which still posts to Slack.

@metadave has a working Datadog monitor for Celery queues in ElasticCache, tested with a full re-render in prod ([bug 1365987](https://bugzilla.mozilla.org/show_bug.cgi?id=1365987)).  He's working on the PR.
